network/src/receiver.rs

```diff
@@ -5,17 +5,19 @@ use crate::error::NetworkError;
 use async_trait::async_trait;
 use bytes::Bytes;
 use futures::stream::{SplitSink, StreamExt as _};
+use futures::SinkExt;
+use std::borrow::BorrowMut;
 use std::{error::Error, net::SocketAddr};
 use tokio::net::{TcpListener, TcpStream};
+use tokio::sync::mpsc::{channel, Receiver as mspcReceiver, Sender};
 use tokio_util::codec::{Framed, LengthDelimitedCodec};
 use tracing::{debug, info, warn};
 
-#[cfg(test)]
-#[path = "tests/receiver_tests.rs"]
-pub mod receiver_tests;
-
 /// Convenient alias for the writer end of the TCP channel.
-pub type Writer = SplitSink<Framed<TcpStream, LengthDelimitedCodec>, Bytes>;
+pub type TcpWriter = SplitSink<Framed<TcpStream, LengthDelimitedCodec>, Bytes>;
+
+/// Convenient alias for the output of the dispatch function.
+pub type Writer = Sender<Bytes>;
```

Comment:
The channel for sending back the handler's response is a full-fleged channel. As long as we do not have the ability to stream to a TCP connection (specifically, we do not have multiplexing capabilities), does this make sense? I.o.w. if we were to fire off two handlers and they each could send a stream of repsonse messages of size > 1, do we expect the other end to make sense of this?

Concretely: should we make this a `oneshot::channel` until we have some multiplexing capability?